src/lib.rs

```diff
@@ -71,8 +71,42 @@ use serde::{de::DeserializeOwned, Serialize};
 use std::fmt;
 use std::sync::Arc;
 
-/// Represents public signature key.
-pub type PublicSignKey = [u8; 32];
+const PUBLIC_ENCRYPT_KEY_BYTES: usize = box_::PUBLICKEYBYTES;
+const SECRET_ENCRYPT_KEY_BYTES: usize = box_::SECRETKEYBYTES;
+const PUBLIC_SIGN_KEY_BYTES: usize = sign::PUBLICKEYBYTES;
+const SECRET_SIGN_KEY_BYTES: usize = sign::SECRETKEYBYTES;
+const SHARED_SECRET_KEY_BYTES: usize = box_::PRECOMPUTEDKEYBYTES;
+const SIGNATURE_BYTES: usize = sign::SIGNATUREBYTES;
+const SYMMETRIC_KEY_BYTES: usize = secretbox::KEYBYTES;
+
+quick_error! {
+    /// This error is returned if encryption or decryption fails.
+    /// The encryption failure is rare and mostly connected to serialisation failures.
+    /// Decryption can fail because of invalid keys, invalid data, or deserialisation failures.
+    #[derive(Debug)]
+    pub enum Error {
+        /// Occurs when serialisation or deserialisation fails.
+        Serialisation(e: SerialisationError) {
+            display("error serialising or deserialising message: {}", e)
+            cause(e)
+            from()
+        }
+        /// Occurs when we can't decrypt a message or verify the signature.
+        DecryptVerify(_e: ()) {
+            display("error decrypting/verifying message")
+            from()
+        }
+        /// Occurs in case of an error during initialisation.
+        InitError(e: i32) {
+            display("error while initialising random generator")
+            from()
+        }
+        /// Occurs when we fail to derive encryption key from password.
+        DeriveKey {
+            display("error deriving encryption key from password")
+        }
+    }
+}
 
 /// Initialise random number generator for the key generation functions.
 pub fn init() -> Result<(), ()> {
```

Comment:
I think it'd be better to return an actual `Error` here rather than the unit type - so e.g.
```rust
pub fn init() -> Result<(), Error> {
    crypto_impl::init().map_err(|_| Error::InitError(-1))
}
```